### PR TITLE
Refactor Backend: Merge OrganizationSQL and Organization

### PR DESCRIPTION
**Pull request by fm3**
### URL of deployed dev instance (used for testing):
- https://refactorbackendorga.webknossos.xyz

### Steps to test:
- register as new user, activate, should be in correct organization team
- test onboarding flow as well, incl. creating organization

### Issues:
- contributes to #2521 

------
- [x] ~~Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- [x] ~~Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- [x] Ready for review

### CODE REVIEW

File: app/models/team/Organization.scala [hunk 1]
```diff
@@ -22,20 +19,21 @@ case class OrganizationSQL(
...
-    )
+  def teamIds(implicit ctx: DBAccessContext): Fox[List[ObjectId]] =
+    TeamSQLDAO.findAllIdsByOrganization(_id)
+
+  def publicWrites(implicit ctx: DBAccessContext): Fox[JsObject] = {
+    Fox.successful(Json.obj(
+      "id" -> _id.toString,
+      "name" -> name,
+      "additionalInformation" -> additionalInformation,
+      "displayname" -> displayName
```
Comment (fm3) [inline app/models/team/Organization.scala]: note to self: @philippotto spoke about renaming this to `displayName` in the json as well, needs to be done here, too

**Comment by philippotto** [comment]
> @fm3 How can I test the onboarding flow?

If you are logged in as scmboy you can open `/onboarding` and follow the steps there.

**Comment by jstriebel** [comment]
@philippotto I get `You are not allowed to create a new organization` when being logged in as scmboy as well as when being logged out.

**Comment by fm3** [comment]
@jstriebel that’s right, the dev servers allow only one organization per instance. this can only be tested locally by setting insertInitialData to false, refreshing schema, and then starting wK. I’m testing it again right now, and I think we’re good to go if I don’t experience errors there. But if you want to test it also, feel free :)

**Comment by philippotto** [comment]
> @jstriebel that’s right, the dev servers allow only one organization per instance. 

But isn't scmboy a super-admin by default? Or is this only true for local development?

**Comment by fm3** [comment]
that setting is overwritten to false on dev

**Pull request merged**